### PR TITLE
Added validate_records in POSTENHANCE_RECORD (INSPIR-1700)

### PR DESCRIPTION
**Pull request by pazembrz**

### CODE REVIEW

**Comment by pazembrz** [comment]
ci test please

File: tests/integration/workflows/test_arxiv_workflow.py [hunk 1]
```diff
@@ -1121,3 +1124,133 @@ def test_workflows_halts_on_multiple_exact_matches(workflow_app):
 
     assert obj.status == ObjectStatus.HALTED
     assert obj.extra_data['_action'] == 'resolve_multiple_exact_matches'
+
+
+def fake_validation(data, schema=None):
+    return
```
Comment (drjova) [inline tests/integration/workflows/test_arxiv_workflow.py]: why we need that?

Comment (pazembrz) [reply tests/integration/workflows/test_arxiv_workflow.py]: To skip first validation and check if second is failing as it should.

File: tests/integration/workflows/test_arxiv_workflow.py [hunk 2]
```diff
@@ -1121,3 +1124,133 @@ def test_workflows_halts_on_multiple_exact_matches(workflow_app):
...
+        'document_type': ['article'],
+        '_collections': ['Literature'],
+        'acquisition_source': {'source': 'arXiv'},
+    }
+    workflow = build_workflow(RECORD_WITH_VALIDATION_ERROR)
+    with requests_mock.Mocker() as requests_mocker:
+        batch_uploader = re.compile('http://localhost:1234/batchuploader/robotupload/*')
+        holidingpen = re.compile('http://test-indexer:9200/holdingpen-hep/hep/*')
+        records = re.compile("http://test-indexer:9200/records-hep/hep/*")
+        ticket = re.compile("http://rt.inspire/ticket/new*")
```
Comment (drjova) [inline tests/integration/workflows/test_arxiv_workflow.py]: do we really need to mock all these requests?

Comment (pazembrz) [reply tests/integration/workflows/test_arxiv_workflow.py]: Yes, we need response 200 from them. I started workflow and I was catching all requests which appeared and was making problems.

**Comment by drjova** [comment]
tests/integration/workflows/test_arxiv_workflow.py:32:1: F401 'requests_mock' imported but unused

**Comment by pazembrz** [comment]
Fixed : tests/integration/workflows/test_arxiv_workflow.py:32:1: F401 'requests_mock' imported but unused

**Pull request merged**